File: C++/reverseString.cpp
// O(N)
// O(1)

#include <vector>

class Solution {
public:
    void reverseString(std::vector<char>& s) {

        if(s.empty()) return;

        int L = 0, R = s.size() -1;

        while( L < R)
        {
            char temp = s[L];
            s[L] = s[R];
            s[R] = temp;
            L++;
            R--;
        }
    }
};

File: Go/reverseString.go
package main

//O(N)
//O(1)
func reverseString(s []byte) {
	if len(s) == 0 {
		return
	}

	L := 0
	R := len(s) - 1

	for L < R {

		s[L], s[R] = s[R], s[L]

		L++
		R--
	}
}
